File: CS/Mvc_DbStorage_Sample/PredefinedReports/ReportsFactory.cs
using DevExpress.XtraReports.UI;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Mvc_DbStorage_Sample.PredefinedReports
{
    public static class ReportsFactory
    {
        static ReportsFactory ()
        {
            Reports.Add("TestReport", () => new TestReport());
        }
        public static Dictionary<string, Func<XtraReport>> Reports = new Dictionary<string, Func<XtraReport>>();
    }
}

File: VB/Mvc_DbStorage_Sample_VB/PredefinedReports/ReportsFactory.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Threading.Tasks
Imports DevExpress.XtraReports.UI

Namespace PredefinedReports
    Public Module ReportsFactory
        Public Reports As New Dictionary(Of String, Func(Of XtraReport))() From {
            { "TestReport", Function() New TestReport() }
        }
    End Module
End Namespace
